packages/cli-repl/src/setup-logger-and-telemetry.ts

```diff
@@ -355,7 +372,10 @@ export default function setupLoggerAndTelemetry(
   });
 
   bus.on('mongosh-sp:connect-heartbeat-failure', function(ev: SpConnectHeartbeatFailureEvent) {
-    log.info('mongosh-sp:connect-heartbeat-failure', ev);
+    log.info('mongosh-sp:connect-heartbeat-failure', {
+      ...ev,
+      failure: ev.failure?.message
+    });
```

Comment:
This is more of a drive-by fix, it matches what we do below. Otherwise, `Error` instances are always just serialized to `{}` (regardless of the rest of this patch).